### PR TITLE
Add translation logic for logstash output

### PR DESCRIPTION
**Pull request by khushijain21**
## What does this PR do?
This PR adds support for logstash output in OTel runtime

## Why is it important?
To support all available outputs in OTel runtime

## Checklist

- [x] I have read and understood the [pull request guidelines](https://github.com/elastic/elastic-agent/blob/main/CONTRIBUTING.md#pull-request-guidelines) of this project.
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Disruptive User Impact

None

## How to test this PR locally
Start a local logstash server on port 5044 that prints events to console

Configure elastic-agent with following config
```yml
outputs:
  default:
    type: logstash
    hosts: ["localhost:5044"]

agent.internal:
  runtime:
    output:
      logstash: otel
  

agent.grpc.port: 6799
inputs:
  - type: system/metrics
    id: unique-system-metrics-input
    data_stream.namespace: default
    use_output: default
    streams:
      - metricsets:
        - cpu
        data_stream.dataset: system.cpu
      - metricsets:
        - memory
        data_stream.dataset: system.memory
      - metricsets:
        - network
        data_stream.dataset: system.network
      - metricsets:
        - filesystem
        data_stream.dataset: system.filesystem

agent.monitoring.enabled: false
```

Watch events streaming on logstash stdout
 

## Related issues

- https://github.com/elastic/elastic-agent/issues/11582

### CODE REVIEW

**Comment by khushijain21** [comment]
Do we need an e2e test here for logstash? There are extensive tests in beats already done [here](https://github.com/elastic/beats/blob/main/x-pack/filebeat/tests/integration/otel_lsexporter_test.go#L62)
We are only translating the config and testing it via UT.

**Comment by leehinman** [comment]
> Do we need an e2e test here for logstash? There are extensive tests in beats already done [here](https://github.com/elastic/beats/blob/main/x-pack/filebeat/tests/integration/otel_lsexporter_test.go#L62)
> We are only translating the config and testing it via UT.

I don't think so.  The beats repo already tests that the docs are equivalent.  Overall we might want an integration test for logstash but I don't think we should hold this PR up for that.

**Review by leehinman — APPROVED** [review]
LGTM.

**Comment by cmacknz** [comment]
Created https://github.com/elastic/elastic-agent/issues/12823 to track adding a configuration option to allows users to easily force inputs using the LS output back to the process runtime.

File: internal/pkg/otel/translate/common.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+// Helper function to conditionally add fields to the map
+func setIfNotNil(m map[string]any, key string, value any) {
+	if value == nil {
+		return
+	}
+
+	v := reflect.ValueOf(value)
+
+	switch v.Kind() {
```
Comment (swiatekm) [inline internal/pkg/otel/translate/common.go:28]: For this logic, I'd do the check itself in a separate function with a switch. We do the same thing in each branch here. The logic for checking this can probably be more consolidated, too - you check length for collection and compare against Zero values for the rest.

I wouldn't be surprised if we could just import this from a dependency.

File: internal/pkg/otel/translate/otelconfig_test.go [hunk 2]
```diff
@@ -2,6 +2,8 @@
 // or more contributor license agreements. Licensed under the Elastic License 2.0;
 // you may not use this file except in compliance with the Elastic License 2.0.
 
+//go:build !requirefips
```
Comment (swiatekm) [inline internal/pkg/otel/translate/otelconfig_test.go]: What's the specific reason we need to do this?

Comment (khushijain21) [reply internal/pkg/otel/translate/otelconfig_test.go]: Reverting this.

Comment (swiatekm) [reply internal/pkg/otel/translate/otelconfig_test.go]: I wonder if we should put that in its own file. Skipping all the config tests in fips mode doesn't sound safe in the long run.

**Comment by cmacknz** [comment]
Created https://github.com/elastic/elastic-agent/issues/13019 to track creating a basic integration test against real Logstash. Agree with Lee, this can be done separately from this PR.

**Comment by cmacknz** [comment]
When https://github.com/elastic/beats/pull/49169 merges, likely we will want a test that setting `status_reporting.enabled: false` that was introduced in https://github.com/elastic/elastic-agent/pull/10890 on the logstash output causes us to correctly mute that status updates.

**Review by leehinman — DISMISSED** [review]
LGTM

Optional suggestion - might be worthwhile to add a second test case that adds a few more edge cases, like hosts and ssl settings that can be a string instead of a list of strings.  I know most of that is in common, so it is tested with ES but just in case something diverges in the future it might be nice to have tests here as well.
